### PR TITLE
Disagg: refresh expiration time of snapshot when calling getSnapshot

### LINKED ISSUE #9567: workload report error “Check snap != nullptr failed: Can not find disaggregated task” when run ch
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1、run ch q7
2、workload report error 

### 2. What did you expect to see? (Required)
no error

### 3. What did you see instead (Required)
workload report error 
[2024-10-28 21:11:43] execute run failed, err execute query q7 failed Error 1105: other error for mpp stream: Code: 11004, e.displayText() = DB::Exception: Check snap != nullptr failed: Can not find disaggregated task, task_id=DisTaskId<MPP<gather_id:1, query_ts:1730149536627187623, local_query_id:5634, server_id:316, start_ts:453548320125354066,task_id:4>,executor=TableFullScan_78> (from s1006_t2737_4307_1_36486282018035081_360582), e.what() = DB::Exception,\r\n"]

### 4. What is your TiFlash version? (Required)
/tiflash/tiflash version
 TiFlash
Release Version: v8.4.0-alpha-80-g08535236b
Edition:         Community
Git Commit Hash: 08535236b8f3d6d41656f22ebb8c2334f3081d86
Git Branch:      HEAD
UTC Build Time:  2024-10-28 11:09:41
Enable Features: jemalloc sm4(GmSSL) mem-profiling avx2 avx512 unwind thinlto hnsw.l2=skylake hnsw.cosine=skylake vec.l2=skylake vec.cos=skylake
Profile:         RELWITHDEBINFO
Compiler:        clang++ 17.0.6

Raft Proxy
Git Commit Hash:   c3bedc86b1470ceacd15dfb19fa7bfa94e8ab49d
Git Commit Branch: HEAD
UTC Build Time:    ""   
Rust Version:      rustc 1.77.0-nightly (89e2160c4 2023-12-27)
Storage Engine:    tiflash
Prometheus Prefix: tiflash_proxy_
Profile:           release
Enable Features:   external-jemalloc portable sse test-engine-kv-rocksdb test-engine-raft-raft-engine openssl-vendored

### PR DESCRIPTION
**Pull request by JinheLin**
### What problem does this PR solve?

Issue Number: close #9567

Problem Summary:

After v7.5, TiFlash use read thread pool to read segments in disagg-mode and the segment will fetch pages only when it is going to be read.

If the query is slow, there is a high probability that the snapshot will expire before the query ends.

### What is changed and how it works?

Refreshing the expiration time of snapshot when calling `getSnapshot`.
```commit-message

```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp [hunk 1]
```diff
@@ -1218,8 +1218,8 @@ void DAGStorageInterpreter::buildLocalStreams(DAGPipeline & pipeline, size_t max
         auto snaps = context.getSharedContextDisagg()->wn_snapshot_manager;
         const auto & snap_id = *dag_context.getDisaggTaskId();
         auto timeout_s = context.getSettingsRef().disagg_task_snapshot_timeout;
-        auto expired_at = Clock::now() + std::chrono::seconds(timeout_s);
```
Comment (Lloyd-Pottiger) [inline dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp:1221]: `disagg_task_snapshot_timeout` can be smaller now?

**Review by CalvinNeo — APPROVED** [review]
lgtm

**Review by Lloyd-Pottiger — APPROVED** [review]

**Review by JaySon-Huang — APPROVED** [review]

**Pull request merged**